### PR TITLE
Better creating GitHub issues

### LINKED ISSUE #599: Create issues for RoP when CLI runs with a config
https://github.com/SAP/fosstars-rating-core/blob/master/src/main/java/com/sap/oss/phosphor/fosstars/tool/github/Application.java#L435

Currently, the CLI creates issues only if it was called with `--url`. It should also create issues with `--config` is used.

### LINKED ISSUE #603: CLI doesn't print reports if --config is used
Looks like a regression after the recent refactoring.

### LINKED ISSUE #614: GitHubDataFetcher.gitHubIssuesFor() should search in a specified project
https://github.com/SAP/fosstars-rating-core/blob/90153f1a40672d0f1de000d1e56869915ca284b4/src/main/java/com/sap/oss/phosphor/fosstars/data/github/GitHubDataFetcher.java#L239

Currently, the method searches for issues in the entire GitHub. Instead, it should use `repo:...` in the query to search only in a specified project.

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here are main updates:

- Updated the CLI to create issues for RoP violations if a config is used (`--config` command-line option). This closes #599
- Updated the CLI to create reports if a config is used. This closes #603
- Updated the GitHub data fetcher to search for issues in a specific repositories. This closes #614
- Fixed several bugs in the Markdown reporter for RoP.

I tested manually that the CLI creates issues for RoP violations:

- Created two projects for testing
  - https://github.com/artem-smotrakov/test-rop-one
  - https://github.com/artem-smotrakov/test-rop-two
- Updated [workflow](https://github.com/artem-smotrakov/fosstars-rating-core/blob/fosstars-oss-rules-of-play-test-workflow/.github/workflows/fosstars-rules-of-play-test.yml) for testing.
- Config and report are stored in [this branch](https://github.com/artem-smotrakov/fosstars-rating-core/tree/fosstars-oss-rules-of-play-test-report)

### CODE REVIEW

**Review by SebastianWolf-SAP — CHANGES_REQUESTED** [review]
Really very elegant solution. I though I included the repository in the search, but obviously I didn't. That's embarrassing. :( 

Anyway, the refactoring and the new solution via a dedicated issue reporter work well - with one exception: The command-line argument is evaluated exactly the wrong way around. That should probably be fixed... ;)

File: src/main/java/com/sap/oss/phosphor/fosstars/tool/OssRulesOfPlayRatingHandler.java [hunk 1]
```diff
@@ -69,7 +70,9 @@ void processMaven(String coordinates) {
   void processUrl(String url) throws IOException {
     GitHubProject project = GitHubProject.parse(url);
     process(project);
-    createIssuesIfRequested(project, commandLine);
+    if (!commandLine.hasOption("create-issues")) {
```
[the remark this answers is not in the document]
Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/tool/OssRulesOfPlayRatingHandler.java:73]: Oops! Good catch!

**Comment by artem-smotrakov** [comment]
> Really very elegant solution. I though I included the repository in the search, but obviously I didn't. That's embarrassing. :(
> 
> Anyway, the refactoring and the new solution via a dedicated issue reporter work well - with one exception: The command-line argument is evaluated exactly the wrong way around. That should probably be fixed... ;)

Thanks for the review and feedback @SebastianWolf-SAP ! I've fixed the bug you found.

**Review by SebastianWolf-SAP — APPROVED** [review]
Thanks for the fix - as I haven't found anything else on Friday and the tiny one is fixed, LGTM now ;)

**Pull request merged**